File: backend/app/schemas/order.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List, Dict
import uuid

class OrderBase(BaseModel):
    """Базова схема замовлення з географічною та фінансовою валідацією."""
    latitude: float = Field(..., ge=-90, le=90, description="Широта (від -90 до 90)")
    longitude: float = Field(..., ge=-180, le=180, description="Довгота (від -180 до 180)")
    subtotal: float = Field(..., gt=0, description="Сума замовлення (має бути > 0)")

class OrderCreate(OrderBase):
    pass

class TaxBreakdown(BaseModel):
    """Деталізація розрахунку складових податку."""
    state_rate: float = 0.0
    county_rate: float = 0.0
    city_rate: float = 0.0
    special_rates: float = 0.0

class OrderResponse(OrderBase):
    """Схема відповіді API з розрахованими податками для замовлення."""
    id: str
    timestamp: datetime
    composite_tax_rate: float
    tax_amount: float
    total_amount: float
    breakdown: Optional[TaxBreakdown]
    jurisdictions: List[str] = []

    class Config:
        from_attributes = True

class PaginatedOrdersResponse(BaseModel):
    """Схема пагінованої відповіді з агрегованою статистикою для дашборду."""
    total: int
    total_tax: float
    avg_rate: float
    page: int
    limit: int
    items: List[OrderResponse]

File: frontend/src/types/order.ts
export interface TaxBreakdown {
  state_rate: number;
  county_rate: number;
  city_rate: number;
  special_rates: number;
}

export interface Order {
  id?: string; // Змінили на string, бо бекенд тепер генерує UUID
  latitude: number;
  longitude: number;
  subtotal: number;
  composite_tax_rate?: number;
  tax_amount?: number;
  total_amount?: number;
  breakdown?: TaxBreakdown;  // Додали нове поле
  jurisdictions?: string[];  // Додали нове поле
}

export interface ImportCSVResponse {
  total_processed: number;
  success_count: number;
  error_count: number;
  errors: {
    row: number;
    reason: string;
  }[];
}
export interface ValidationError {
  loc: (string | number)[];
  msg: string;
  type: string;
}

export interface ApiError {
  detail: string | ValidationError[]; // FastAPI повертає або рядок, або масив деталей
}
// src/types/order.ts
export interface LoginPayload {
  email: string;
  password: string;
}

export interface LoginResponse {
  access_token: string;       // JWT-токен
  expires_in?: number; // необов'язково, якщо бекенд віддає час життя токена
  user?: {
    id: string;
    email: string;
    name?: string;
  };}